cmds/core/tee/tee.go

```diff
@@ -25,53 +25,70 @@ import (
 	flag "github.com/spf13/pflag"
 )
 
-const name = "tee"
-
 var (
 	cat    = flag.BoolP("append", "a", false, "append the output to the files rather than rewriting them")
 	ignore = flag.BoolP("ignore-interrupts", "i", false, "ignore the SIGINT signal")
 )
 
-// handeFlags parses all the flags and sets variables accordingly
-func handleFlags() int {
-	flag.Parse()
+type command struct {
+	stdin  io.Reader
+	stdout io.Writer
+	stderr io.Writer
+	cat    bool
+	ignore bool
+	args   []string
+}
 
-	oflags := os.O_WRONLY | os.O_CREATE
+func tee(cat, ignore bool, args []string) *command {
```

Comment:
It is idiomatic in cases like this to call the function new, if you can.

Reply:
Sure, but it will be New to avoid name collision with builtin function.